packages/rum-core/src/common/apm-server.js

```diff
@@ -207,25 +206,32 @@ class ApmServer {
   }
 
   ndjsonErrors(errors) {
-    return errors.map(function(error) {
-      return NDJSON.stringify({ error })
-    })
+    return errors.map(error => NDJSON.stringify({ error }))
+  }
+
+  ndjsonMetricsets(metricsets) {
+    const timestamp = Date.now() * 1000
+    return metricsets
+      .map(metricset =>
+        NDJSON.stringify({ metricset: { timestamp, ...metricset } })
+      )
+      .join('')
   }
 
   ndjsonTransactions(transactions) {
-    var ndjsonSpan = this.ndjsonSpan
```

Comment:
This was a small optimisation to avoid creating new new objects every time in the spans loop

Reply:
In most of the browsers I tested, some via saucelabs. new object is faster and its a small micro optimization so we dont need to worry about it much. 
https://jsperf.com/new-object-vs-override